rustic check --read-data doesn't work on hot+cold repos

Hey all, I'm trying to migrate to a hot/cold repo using Oracle Cloud Object Storage from an existing Restic repo. I've got two buckets, restic-hot and restic-cold.
1. I warmed up everything in restic-cold. 
2. I ran `rustic init --hot-only` and it copied over the files successfully
3. I wanted to check the data integrity of the repo itself so I ran `rustic check --read-data --read-data-subset=1/25` this generated a bunch of errors as it appears to be trying to read the pack data from the hot repo and failed
```
[WARN] service=s3 name=restic-hot path=data/94/94f5744d9ef6f752579a12ee4eaf853910c6e73b3715e7e6f4cf09a6c8379abf: read failed NotFound (persistent) at read, context: { uri: https://<namespace>.compat.objectstorage.us-sanjose-1.oraclecloud.com/restic-hot/restic_media/data/94/94f5744d9ef6f752579a12ee4eaf853910c6e73b3715e7e6f4cf09a6c8379abf, response: Parts { status: 404, ... }, service: s3, path: data/94/94f5744d9ef6f752579a12ee4eaf853910c6e73b3715e7e6f4cf09a6c8379abf, range: 0- } => S3Error { code: "NoSuchKey", message: "The object 'restic_media/data/94/94f5744d9ef6f752579a12ee4eaf853910c6e73b3715e7e6f4cf09a6c8379abf' was not found in the bucket 'restic-hot'", resource: "", request_id: "" }

[ERROR] error reading pack 94f5744d : `rustic_core` experienced an error related to `the backend`.

Message:
Reading file `data/94/94f5744d9ef6f752579a12ee4eaf853910c6e73b3715e7e6f4cf09a6c8379abf` failed in the backend. Please check if the given path is correct.

Context:
- type: Pack
- id: 94f5744d

Caused by:
NotFound (persistent) at read, ... => S3Error { code: "NoSuchKey", message: "The object 'restic_media/data/94/...' was not found in the bucket 'restic-hot'", ... }
```

I was able to confirm that file restores do work correctly. Seems to be a specific issue to the check-data command

--- a/tests/check_hot_cold.rs
+++ b/tests/check_hot_cold.rs
@@ -1,0 +1,51 @@
+use std::path::Path;
+
+use assert_cmd::Command;
+use rustic_testing::TestResult;
+use tempfile::tempdir;
+
+fn rustic_runner(profile: &Path, cold_repo: &Path, hot_repo: &Path) -> TestResult<Command> {
+    let mut runner = Command::new(env!("CARGO_BIN_EXE_rustic"));
+    runner
+        .arg("--use-profile")
+        .arg(profile)
+        .arg("--repository")
+        .arg(cold_repo)
+        .arg("--repo-hot")
+        .arg(hot_repo)
+        .args(["--password", "test", "--no-progress", "--no-cache"]);
+    Ok(runner)
+}
+
+#[test]
+fn check_read_data_uses_cold_data_packs_for_hot_cold_repositories() -> TestResult<()> {
+    let temp_dir = tempdir()?;
+    let cold_repo = temp_dir.path().join("cold-repository");
+    let hot_repo = temp_dir.path().join("hot-repository");
+    let source = temp_dir.path().join("source");
+    let profile = temp_dir.path().join("test.toml");
+    std::fs::write(&profile, "")?;
+    std::fs::create_dir(&source)?;
+    std::fs::write(source.join("payload.bin"), vec![42_u8; 256 * 1024])?;
+
+    rustic_runner(&profile, &cold_repo, &hot_repo)?
+        .arg("init")
+        .assert()
+        .success();
+
+    rustic_runner(&profile, &cold_repo, &hot_repo)?
+        .arg("backup")
+        .arg(&source)
+        .assert()
+        .success();
+
+    // Data packs are intentionally stored only in the cold repository. A
+    // successful read-data check proves it reads them there rather than trying
+    // the hot metadata repository.
+    rustic_runner(&profile, &cold_repo, &hot_repo)?
+        .args(["check", "--read-data"])
+        .assert()
+        .success();
+
+    Ok(())
+}

